internal/setup/setup.go

```diff
@@ -534,14 +535,34 @@ func parsePackage(baseDir, pkgName, pkgPath string, data []byte) (*Package, erro
 			Scripts: SliceScripts{
 				Mutate: yamlSlice.Mutate,
 			},
+			Essential: map[SliceKey]bool{},
+		}
+		for _, refName := range yamlPkg.Essential {
+			sliceKey, err := ParseSliceKey(refName)
+			if err != nil {
+				return nil, fmt.Errorf("invalid slice reference %q in %s", refName, pkgPath)
+			}
+			if sliceKey.Package == slice.Package && sliceKey.Slice == slice.Name {
+				// Do not add the slice to its own essentials list.
+				continue
+			}
+			if _, ok := slice.Essential[sliceKey]; ok {
+				return nil, fmt.Errorf("cannot reference slice %q twice in essentials for slice %q in %s", refName, slice.String(), pkgPath)
```

Comment:
"slice %s lists essential redundantly: %s", (refName, slice)

Here we also don't need the %q quoting, because that'll necessarily contain an underline as it's already been validated, which makes it unambiguous within the sentence. We also shouldn't need String() if it's an %s.